### PR TITLE
Feature/5972 index downloaded data

### PR DESCRIPTION
**Pull request by HavierD**
Doesn't include "indexAll" api yet;
Doesn't have unit tests for new added features yet;

### CODE REVIEW

File: indexer/src/main/java/au/org/aodn/esindexer/configuration/AppConstants.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 public interface AppConstants {
     String PORTAL_RECORDS_MAPPING_JSON_FILE = "portal_records_index_schema.json";
+    String DTASET_INDEX_MAPPING_JSON_FILE = "dataset_index_schema.json";
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/configuration/AppConstants.java:5]: Spelling dtaser

File: indexer/src/main/java/au/org/aodn/esindexer/configuration/DatasetAccessConfig.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import au.org.aodn.esindexer.service.DatasetAccessServiceImpl;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class DatasetAccessConfig {
+
+    @Bean(name = "DatasetAccessService")
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/configuration/DatasetAccessConfig.java:11]: The repo is data-access-service, maybe you call it the same way, DataAccessService, be less varied in names.

File: indexer/src/main/java/au/org/aodn/esindexer/configuration/DatasetAccessConfig.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    public  DatasetAccessServiceImpl createDatasetAccessService(
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/configuration/DatasetAccessConfig.java:12]: Same, dataset to data.

File: indexer/src/main/java/au/org/aodn/esindexer/configuration/DatasetAccessConfig.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        @Value("${dataaccess.host:defaulthost}") String serverUrl
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/configuration/DatasetAccessConfig.java:13]: dataAccessSevice.host.default

Host then also defaultHost, not necessary

File: indexer/src/main/java/au/org/aodn/esindexer/controller/IndexerController.java [hunk 5]
```diff
@@ -33,6 +38,9 @@ public class IndexerController {
     @Autowired
     GeoNetworkService geonetworkResourceService;
 
+    @Autowired
+    DatasetAccessService datasetAccessService;
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/controller/IndexerController.java:42]: dataAccessService

File: indexer/src/main/java/au/org/aodn/esindexer/service/DatasetAccessService.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package au.org.aodn.esindexer.service;
+
+import au.org.aodn.esindexer.model.Dataset;
+
+import java.time.LocalDate;
+
+public interface DatasetAccessService {
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/service/DatasetAccessService.java:7]: DataAccessService

File: indexer/src/main/java/au/org/aodn/esindexer/model/DataRecord.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.time.LocalDate;
+
+// TODO: if more fields are needed to be filtered, please add more columns here
+@Getter
+@Setter
+public class DataRecord{
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/model/DataRecord.java:11]: I think they all need to reconsider a clearer and consistent name, to be less confusing,

As is growing, the terms 
: data record, metadata record, data, dataset etc. 

What are they? Are they the same thing or different?

File: indexer/src/main/java/au/org/aodn/esindexer/service/DatasetAccessServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+            params.put("uuid", uuid);
+
+            String url = UriComponentsBuilder.fromHttpUrl(getDatasetAccessEndpoint() + "/data/{uuid}")
+                    .queryParam("is_to_index", "true")
+                    .queryParam("start_date", startDate)
+                    .queryParam("end_date", endDate)
+                    .buildAndExpand(uuid)
+                    .toUriString();
+
+            ResponseEntity<DataRecord[]> responseEntity = restTemplate.exchange(
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/service/DatasetAccessServiceImpl.java:58]: DataRecord is a Dto? Then you should rename the model class to make it clear, with Dto postfix.

Comment (vietnguyengit) [reply indexer/src/main/java/au/org/aodn/esindexer/service/DatasetAccessServiceImpl.java:58]: Get....Dto, eg GetDataRecordDto

Create....Dto, e.g CreateDataRecordDto

**Review by vietnguyengit — CHANGES_REQUESTED** [review]
Just naming changes overall, the choice of changes are up to you, must make sure the convention is consistent and clear and be less varied.

File: indexer/src/main/java/au/org/aodn/esindexer/service/DatasetAccessServiceImpl.java (hunk 8 above)
Comment (HavierD) [reply indexer/src/main/java/au/org/aodn/esindexer/service/DatasetAccessServiceImpl.java:58]: ah, it is not a standard DTO. Maybe i should rename `DataRecord` into `AggregatedDatum`, to express it is a single datum of a dataset with aggregated count

Comment (HavierD) [reply indexer/src/main/java/au/org/aodn/esindexer/service/DatasetAccessServiceImpl.java:58]: or maybe just `Datum`?

**Review by vietnguyengit — APPROVED** [review]

**Pull request merged**